includes/login-register.php

```diff
@@ -210,17 +216,37 @@ function edd_login_register_error_message( $errors, $redirect ) {
  * @return string
  */
 function edd_get_lostpassword_url() {
-	$url      = wp_validate_redirect( edd_get_current_page_url(), edd_get_checkout_uri() );
-	$redirect = add_query_arg(
+
+	return add_query_arg(
 		array(
-			'checkemail'         => 'confirm',
-			'edd_reset_password' => 'confirm',
-			'edd_redirect'       => urlencode( $url ),
+			'edd_forgot_password' => 'confirm',
 		),
-		wp_login_url()
+		wp_lostpassword_url(
```

Comment:
@cklosowski I'm kind of inclined to remove the redirect parameter from the `wp_lostpassword_url()` call here and just manage the login errors filter based solely on the `edd_forgot_password_redirect` being available in the session. What would you think of that? It lets WP Core handle the redirect_to automatically instead of us forcing it to be the login URL.